code/game/objects/items/weapons/weaponry.dm

```diff
@@ -320,11 +320,58 @@
 	throwforce = 12
 	attack_verb = list("beat", "smacked")
 	w_class = 5
+	var/homerun_ready = 0
+	var/homerun_able = 0
+
+/obj/item/weapon/melee/baseball_bat/homerun
+	name = "home run bat"
+	desc = "This thing looks dangerous... Dangerously good at baseball, that is."
+	homerun_able = 1
+
+/obj/item/weapon/melee/baseball_bat/attack_self(mob/user)
+	if(!homerun_able)
+		..()
+		return
+	if(homerun_ready)
+		user << "<span class='notice'>You're already ready to do a home run!</span>"
+		..()
+		return
+	user << "<span class='warning'>You begin gathering strength...</span>"
+	playsound(get_turf(src), 'sound/magic/lightning_chargeup.ogg', 65, 1)
+	if(do_after(user, 90, target = src))

```

Comment:
Not sure this is going to work; target may need to be user to get the bar to show up.